src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -1159,26 +1173,31 @@ def add_units_table(
     excluded_properties = list(skip_properties) + ["contact_vector"]
     properties_to_extract = [property for property in sorting_properties if property not in excluded_properties]
 
+    if unit_ids is not None:
+        checked_sorting = checked_sorting.select_units(unit_ids=unit_ids)
+    unit_ids = checked_sorting.unit_ids
+
     # Extract properties
     for property in properties_to_extract:
         data = checked_sorting.get_property(property)
+        if isinstance(data[0], (bool, np.bool_)):
+            data = data.astype(str)
         index = isinstance(data[0], (list, np.ndarray, tuple))
         description = property_descriptions.get(property, "No description.")
         data_to_add[property].update(description=description, data=data, index=index)
         if property in ["max_channel", "max_electrode"] and nwbfile.electrodes is not None:
             data_to_add[property].update(table=nwbfile.electrodes)
 
     # Unit name logic
-    units_ids = checked_sorting.get_unit_ids()
     if "unit_name" in data_to_add:
+        # if 'unit_name' is set as a property, it is used to override default unit_ids (and "id")
         unit_name_array = data_to_add["unit_name"]["data"]
     else:
-        unit_name_array = units_ids.astype("str", copy=False)
+        unit_name_array = unit_ids.astype("str", copy=False)
         data_to_add["unit_name"].update(description="Unique reference for each unit.", data=unit_name_array)
-
-    # If the channel ids are integer keep the old behavior of asigning table's id equal to unit_ids
-    if np.issubdtype(units_ids.dtype, np.integer):
-        data_to_add["id"].update(data=units_ids.astype("int"))
+        # If the channel ids are integer keep the old behavior of asigning table's id equal to unit_ids
+        if np.issubdtype(unit_ids.dtype, np.integer):
+            data_to_add["id"].update(data=unit_ids.astype("int"))
```

Comment:
@CodyCBakerPhD same here for units: passing the `unit_name` property overrides the default names assigned by `neuroconv`. However, if `unit_ids` are `int`, the `id` field of the rows is not consistent. In this case, we only set the `id` if the `unit_name` property is not set.